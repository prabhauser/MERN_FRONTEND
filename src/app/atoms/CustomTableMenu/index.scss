@import '../../../utilities/theme/_theme.scss';

.customTableMenuAction{
    color: $p800Bg;
}