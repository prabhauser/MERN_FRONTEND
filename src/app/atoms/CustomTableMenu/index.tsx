import { Icon } from '@iconify/react';
import { useRef, useState } from 'react';
import editFill from '@iconify/icons-eva/edit-fill';
import { Link as RouterLink } from 'react-router-dom';
import trash2Outline from '@iconify/icons-eva/trash-2-outline';
import moreVerticalFill from '@iconify/icons-eva/more-vertical-fill';
import { Menu, MenuItem, IconButton, ListItemIcon, ListItemText } from '@mui/material';
import './index.scss';

interface Props {
    click: any;
    removeFn?: any;
}
export default function UserMoreMenu(props: Props) {
    const { click, removeFn = () => {} } = props;
    const ref = useRef(null);
    const [isOpen, setIsOpen] = useState(false);

    return (
        <>
            <IconButton ref={ref} onClick={() => setIsOpen(true)}>
                <Icon className="customTableMenuAction" icon={moreVerticalFill} width={20} height={20} />
            </IconButton>

            <Menu
                open={isOpen}
                anchorEl={ref.current}
                onClose={() => setIsOpen(false)}
                PaperProps={{
                    sx: { width: 118, maxWidth: '100%', borderRadius: '10px' }
                }}
                anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
                transformOrigin={{ vertical: 'top', horizontal: 'right' }}
                onClick={() => setIsOpen(false)}
            >
                {/* component={RouterLink} to="#" */}
                <MenuItem sx={{ color: 'text.secondary' }} className="table_actionmenu">
                    {/* <ListItemIcon>
                        <Icon icon={editFill} width={24} height={24} />
                    </ListItemIcon> */}
                    <ListItemText primary="View" primaryTypographyProps={{ variant: 'body2' }} onClick={click} />
                </MenuItem>

                <MenuItem sx={{ color: 'text.secondary' }} className="table_actionmenu">
                    {/* <ListItemIcon>
                        <Icon icon={trash2Outline} width={24} height={24} />
                    </ListItemIcon> */}
                    <ListItemText primary="Remove" primaryTypographyProps={{ variant: 'body2' }} onClick={removeFn} />
                </MenuItem>
            </Menu>
        </>
    );
}
