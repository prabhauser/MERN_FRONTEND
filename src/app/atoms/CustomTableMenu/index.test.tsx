import { shallow } from 'enzyme';
import CustomTableMenu from './index';

describe('<CustomTableMenu />', () => {
    it('should take tablemenu snapshot', () => {
        const component = shallow(<CustomTableMenu click={() => {}} />);
        component.update();
        expect(component).toMatchSnapshot();
    });
});
