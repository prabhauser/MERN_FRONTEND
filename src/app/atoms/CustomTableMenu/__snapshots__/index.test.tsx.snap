// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTableMenu /> should take tablemenu snapshot 1`] = `
<Fragment>
  <ForwardRef(IconButton)
    onClick={[Function]}
  >
    <ForwardRef(Icon)
      className="customTableMenuAction"
      height={20}
      icon={
        Object {
          "body": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"2\\" fill=\\"currentColor\\"/><circle cx=\\"12\\" cy=\\"5\\" r=\\"2\\" fill=\\"currentColor\\"/><circle cx=\\"12\\" cy=\\"19\\" r=\\"2\\" fill=\\"currentColor\\"/>",
          "height": 24,
          "width": 24,
        }
      }
      width={20}
    />
  </ForwardRef(IconButton)>
  <ForwardRef(Menu)
    PaperProps={
      Object {
        "sx": Object {
          "borderRadius": "10px",
          "maxWidth": "100%",
          "width": 118,
        },
      }
    }
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    onClick={[Function]}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <ForwardRef(MenuItem)
      className="table_actionmenu"
      sx={
        Object {
          "color": "text.secondary",
        }
      }
    >
      <ForwardRef(ListItemText)
        onClick={[Function]}
        primary="View"
        primaryTypographyProps={
          Object {
            "variant": "body2",
          }
        }
      />
    </ForwardRef(MenuItem)>
    <ForwardRef(MenuItem)
      className="table_actionmenu"
      sx={
        Object {
          "color": "text.secondary",
        }
      }
    >
      <ForwardRef(ListItemText)
        onClick={[Function]}
        primary="Remove"
        primaryTypographyProps={
          Object {
            "variant": "body2",
          }
        }
      />
    </ForwardRef(MenuItem)>
  </ForwardRef(Menu)>
</Fragment>
`;
